.github/workflows/dev_deployment.yml

```diff
@@ -0,0 +1,35 @@
+name: Initiate Dev Environmnet Deployment
+
+on:
+    push: #Add back master branch after testing
+env:
+    DEV_VM_IP: '${{secrets.DEV_VM_IP}}'
+    DEV_VM_PSWD: '${{secrets.DEV_VM_PSWD}}'
+    DISCORD_WH: '${{secrets.DISCORD_WEBHOOK}}'
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                python-version: ["3.10"]
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v4
+              with:
+                python-version: ${{ matrix.python-version }}
+
+            - name: Install dependencies
+              run: |
+                python3 -m pip install --upgrade pip
+                pip3 install paramiko
+
+            - name: Extract branch name
+              id: extract_branch_name
+              run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
```

Comment:
Help me understand this. The workflow's environment has an environment variable called `GITHUB_OUTPUT` pointing to a file where you may store state?

And then you can load the state by accessing the steps' outputs file holding the variable called branch?

Are you sure this is how it's done?

Reply:
yes, this was in the stack overflow link that you send me

Reply:
so the issue might be coming from the fact that the step is missing the `id` property